lib/rules/no-test-file-importing.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileOverview no importing of test files are allowed
+ */
+
+'use strict';
+
+const REPORT_MESSAGE
+  = 'Do not import "-test.js" file in a test file. This will cause the module and tests from the imported file to be executed again.';
+
+/**
+ * Returns the name of the module imported or re-exported.
+ *
+ * @param {ASTNode} node - A node to get.
+ * @returns {string} the name of the module, or empty string if no name.
+ */
+function getValue(node) {
+  if (node && node.source && node.source.value) {
```

Comment:
Would you ever have an `ImportDeclaration` with no source value? Based on [this](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import), it doesn't appear so.

Reply:
Was following the no-duplicate-imports rule (https://github.com/eslint/eslint/blob/fc7f404365e87fb6ee4f30149e486229cda4eb2e/lib/rules/no-duplicate-imports.js) when I added this. But it does make sense that it shouldn't have no source.